### PR TITLE
[Refactor]Refacor query queue

### PR DESCRIPTION
**Pull request by wangbo**
## Proposed changes
Use CompletableFuture to refactor query queue to simplify code for better readability.

### CODE REVIEW

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by liutang123 — DISMISSED** [review]
LGTM

File: fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java [hunk 1]
```diff
@@ -656,11 +656,7 @@ public void exec() throws Exception {
                         throw new UserException("could not find query queue");
                     }
                     queueToken = queryQueue.getToken();
-                    if (!queueToken.waitSignal(this.queryOptions.getExecutionTimeout() * 1000)) {
-                        LOG.error("query (id=" + DebugUtil.printId(queryId) + ") " + queueToken.getOfferResultDetail());
```
Comment (HappenLee) [inline fe/fe-core/src/main/java/org/apache/doris/qe/Coordinator.java:660]: where to log the detail，should we still need to log？

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Comment by wangbo** [comment]
run buildall

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by xinyiZzz — APPROVED** [review]
LGTM

**Pull request merged**